src/csharp/Grpc.Tools/ProtoCompilerOutputs.cs

```diff
@@ -68,8 +76,13 @@ public override bool Execute()
             // Get language-specific possible output. The generator expects certain
             // metadata be set on the proto item.
             var possible = new List<ITaskItem>();
+            var patched = new List<ITaskItem>();
             foreach (var proto in Protobuf)
             {
+                // This operates on a local copy and has no effect on the MSBuild instance!
+                generator.PatchOutputDirectory(proto);
+                patched.Add(proto);
```

Comment:
this seems odd. You're taking items from `Protobuf` and you're modifying them (without making a defensive copy), so once done, both Protobuf and PatchedProtobuf will contain the same instances of ITaskItem, which is very confusing (and the items in Protobuf will all be patched, which is counterintuitive given the naming).

Reply:
`Protobuf` is a local copy of the MSBuild items (see above). I can create copies and assign them to a separate variable to make it more clear, but this will result in duplicating items twice (once from MSBuild into the task and again in PatchOutputDirectory).